rag/llm/ocr_model.py

```
@@ -98,6 +104,210 @@ def parse_pdf(self, filepath: str, binary=None, callback=None, parse_method: str
+        try:
+            r = requests.get(
+                f"{self.base_url}/api/v4/extract/task/connection-check",
+                headers={"Authorization": f"Bearer {self.api_key}"},
+                timeout=10,
+            )
+            if r.status_code in (401, 403):
+                return False, f"authentication failed (HTTP {r.status_code}): {r.text[:200]}"
+            # 404 (task not found) confirms the key is valid and the server is reachable.
+            logging.info("[MinerU.Net] connection check passed, status=%d", r.status_code)
+            return True, ""
+        except Exception as exc:
```

Ruff (0.16.0)

[warning] Do not catch blind exception: `Exception`

(BLE001)